tvb_library/tvb/datasets/tvb_data.py

```diff
@@ -0,0 +1,227 @@
+# -*- coding: utf-8 -*-
+#
+#
+# TheVirtualBrain-Scientific Package. This package holds all simulators, and
+# analysers necessary to run brain-simulations. You can use it stand alone or
+# in conjunction with TheVirtualBrain-Framework Package. See content of the
+# documentation-folder for more details. See also http://www.thevirtualbrain.org
+#
+# (c) 2012-2023, Baycrest Centre for Geriatric Care ("Baycrest") and others
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software Foundation,
+# either version 3 of the License, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+# You should have received a copy of the GNU General Public License along with this
+# program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+#   CITATION:
+# When using The Virtual Brain for scientific publications, please cite it as explained here:
+# https://www.thevirtualbrain.org/tvb/zwei/neuroscience-publications
+#
+#
+
+"""
+.. moduleauthor:: Abhijit Deo <f20190041@goa.bits-pilani.ac.in>
+"""
+
+import os
+import requests
+import json
+import pooch
+from pathlib import Path
+from zipfile import ZipFile
+import shutil
+
+from .base import BaseDataset
+from .zenodo import Zenodo, Record, BASE_URL
+
+class TVBZenodoDataset(BaseDataset):
+
+    CONCEPTID = "3417206"
+    
+    def __init__(self, version= "2.7", extract_dir = None):
+        """
+        Constructor for TVB_Data class 
+
+        parameters
+        -----------
+
+        version: str
+              - Version number of the dataset, Default value is 2.7
+        
+        extract_dir: str
+              - path where you want to extract the archive.
+              - If `extract_dir` is None, Dataset is downloaded at location according to your profile settings.
+
+        """
+        super().__init__(version, extract_dir)
+        self.cached_dir = self.extract_dir / ".cache" 
+        self.cached_file = self.cached_dir / "tvb_cached_responses.txt"
+        self.files_in_zip_dict = None
+
+        if  not self.cached_dir.is_dir():
+            self.cached_dir.mkdir(parents=True)
+
+        try:
+            self.recid = self.read_cached_response()[version]['conceptrecid']
+            
+        except :
+            self.log.warning(f"Failed to read data from cached response.")
+            self.recid = Zenodo().get_versions_info(self.CONCEPTID)[version]            
+            self.update_cached_response()
+         
+        self.rec = Record(self.read_cached_response()[self.version])
+
+    def download(self, path=None, fname=None):
```

Comment:
Having both `path` here and `extract_dir` in the constructor is confusing to me.
Maybe we can have a call to discuss ?